packages/devtools_app/lib/src/debugger/flutter/console.dart

```diff
@@ -3,24 +3,85 @@
 // found in the LICENSE file.
 
 import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
 
 import '../../flutter/common_widgets.dart';
+import '../../flutter/theme.dart';
 import 'common.dart';
+import 'debugger_controller.dart';
+
+// TODO(devoncarew): Allow scrolling horizontally as well.
+
+// TODO(devoncarew): Show some small UI indicator when we receive stdout/stderr.
+
+// TODO(devoncarew): Support hyperlinking to stack traces.
+
+/// Display the stdout and stderr output from the process under debug.
+class Console extends StatefulWidget {
+  const Console({
+    Key key,
+    this.controller,
+  }) : super(key: key);
+
+  final DebuggerController controller;
+
+  @override
+  _ConsoleState createState() => _ConsoleState();
+}
+
+class _ConsoleState extends State<Console> {
+  final scrollController = ScrollController();
 
-class Console extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final theme = Theme.of(context);
+    final textStyle =
+        theme.textTheme.bodyText2.copyWith(fontFamily: 'RobotoMono');
 
     return OutlinedBorder(
       child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
         children: [
           debuggerSectionTitle(theme, text: 'Console'),
-          const Expanded(
-            child: Center(child: Text('todo:')),
+          Expanded(
+            child: Padding(
+              padding: const EdgeInsets.all(denseSpacing),
+              child: SingleChildScrollView(
+                controller: scrollController,
+                child: ValueListenableBuilder(
+                  valueListenable: widget.controller.stdio,
+                  builder: (context, value, _) {
+                    // If we're at the end already, scroll to expose the new
+                    // content.
+                    final pos = scrollController.position;
+                    if (pos.pixels == pos.maxScrollExtent) {
+                      SchedulerBinding.instance.addPostFrameCallback((_) {
+                        _scrollToBottom();
+                      });
+                    }
+
+                    return Text(value, style: textStyle);
+                  },
+                ),
+              ),
+            ),
           ),
         ],
       ),
     );
   }
+
+  void _scrollToBottom() async {
+    await scrollController.animateTo(
+      scrollController.position.maxScrollExtent,
+      duration: const Duration(milliseconds: 100),
+      curve: Curves.linear,
```

Comment:
linear isn't a typical choice for an animation curve. document why it is better hear than the more common easeInEaseOut.

Reply:
see
theme.dart
which has our defaults for durations and curves.
`defaultCurve`
`defaultDuration`
`shortDuration`